narwhals/_pandas_like/expr.py

```diff
@@ -330,3 +342,166 @@ def total_microseconds(self) -> PandasExpr:
 
     def total_nanoseconds(self) -> PandasExpr:
         return register_namespace_expression_call(self._expr, "dt", "total_nanoseconds")
+
+
+class PandasExprNameNamespace:
+    def __init__(self, expr: PandasExpr) -> None:
+        self._expr = expr
+
+    def map(self, function: Callable[[str], str]) -> PandasExpr:
+        root_names = self._expr._root_names
+        implementation = self._expr._implementation
+
+        if root_names is None:
+            msg = (
+                "Anonymous expressions are not supported in `.name.map`.\n"
+                "Instead of `nw.all()`, try using a named expression, such as "
+                "`nw.col('a', 'b')`\n"
+            )
+            raise ValueError(msg)
+
+        def mapper(df: PandasDataFrame) -> list[PandasSeries]:
+            tmp = df.select(root_names)
```

Comment:
I don't think you want to do the select on the original dataframe, otherwise you lose all the other transformations that took place

e.g.:
```python
In [9]: df = pd.DataFrame({'a': [2, 1, 3], 'b': [4, 5, 6]})

In [10]: nw.to_native(nw.from_native(df).with_columns((nw.col('a')*2).name.suffix('_doubled')))
Out[10]:
   a  b  a_doubled
0  2  4          2
1  1  5          1
2  3  6          3
```

Reply:
I see, let me take a better look at this and test it properly as well